index.js

```diff
@@ -29,26 +34,32 @@ function lint(textEditor) {
 		pkg = dir === null ? pkg : loadJson(path.join(dir, 'package.json'));
 	}
 
-	// `pkg.xo === false` && xo is a dependency && there's no parent `package.json`
-	if (dir === null) {
+	return {pkg, dir};
+}
+
+function lint(textEditor) {
+	const filePath = textEditor.getPath();
+	const currentDir = pkgDir.sync(path.dirname(filePath));
+
+	if (currentDir === null) {
 		return [];
 	}
 
-	// only lint when `xo` is a dependency
-	if (!(pkg.dependencies && pkg.dependencies.xo) &&
-		!(pkg.devDependencies && pkg.devDependencies.xo)) {
+	const {pkg, rootDir} = findPkg(currentDir);
+
+	if (!rootDir || rootDir === null || !xoInPkg(pkg)) {
```

Comment:
Why are you doing both `!rootDir` and `rootDir === null`? Seems like the latter should be enough.